### ISSUE TITLE
Add Type Definitions

### ISSUE THREAD

**Issue by jpike88** [caolan/async#1831]
See title.

@types/async definitions are way out of date, would be better for everyone if you guys integrated them into your main project

**Comment by Troublor** [comment]
Agree, it's annoying when encounter outdated type definitions.

**Comment by justarandomgeek** [comment]
Indeed, it fairly effectively prevents any reasonable usage of newer features from TypeScript...

**Comment by aearly** [comment]
I've considered a re-write in TS (or at least adding Typedoc comments to everything and generating a `.d.ts` automatically).  It would be a lot of work though.

**Comment by jeandat** [comment]
Types would be nice indeed.

**Comment by kiprobinsonknack** [comment]
Is `@types/async` generated by you or someone else? e.g. if i notice a specific problem, should i report it as an issue here, or create a PR in the definitely-typed repo?

**Comment by aearly** [comment]
We don't maintain those. It's all in the definitely typed repo.
